pkg/stanza/operator/input/windows/input.go

```diff
@@ -216,6 +216,16 @@ func (i *Input) read(ctx context.Context) int {
 	events, err := i.subscription.Read(i.maxReads)
 	if err != nil {
 		i.Logger().Error("Failed to read events from subscription", zap.Error(err))
+		if i.isRemote() && (err.Error() == "The handle is invalid." || err.Error() == "subscription handle is not open") {
```

Comment:
You will get "subscription handle is not open" because the handle is of a new subscription is only set after successfully opening the subscription.

Is [`EvtNext`](https://learn.microsoft.com/en-us/windows/win32/api/winevt/nf-winevt-evtnext) is returning `The handle is invalid.` when the remote server goes offline? It will be better to match the error returned by the win32 API something like `errors.Is(err, windows.ERROR_INVALID_HANDLE)` instead of the string.

Reply:
Thanks for pointing out the err matching.